basic-linux-keysniffer/io/consts.cpp

```diff
@@ -0,0 +1,7 @@
+//
+// Created by gnir on 6/7/19.
+//
+namespace KeyboardEventReader {
+
+    extern constexpr char fileLocation[] = "/dev/input/event3";
```

Comment:
constexpr expressions are compile time constants, so I think you need to just define it in the header (without the extern).